lib/cocoapods/downloader/cache.rb

```diff
@@ -279,23 +304,58 @@ def in_tmpdir(&blk)
       #
       # @return [Void]
       #
-      def copy_and_clean(source, destination, spec)
-        specs_by_platform = group_subspecs_by_platform(spec)
+      def copy_source_and_clean(source, destination, spec)
+        specs_by_platform = group_subspecs_by_platform([spec])
         destination.parent.mkpath
         Cache.write_lock(destination) do
-          FileUtils.rm_rf(destination)
```

Comment:
I do not understand how we grabbed a lock and then deleted the entire folder here? Was it obviating the lock?

Reply:
Either way I preserved this logic below too.

Reply:
I assume the write lock is just to make mutations exclusive to avoid invalidating other parallel work in the cache.

Reply:
yes but we are actually deleting the folder the lock resides in?

Reply:
Oh, good point, I think you're right, not sure this is doing what we'd expect.

Reply:
Actually, isn't the "#{destination}.lock" sit next to the folder, as opposed to within it?

Reply:
ah yes thats what happens.

Reply:
now that this is figured out I updated the code to work for parallel execution.